### PR TITLE
fix(provider/aws): Prevent NullPointerException when image id is not in the cache

### PR DESCRIPTION
**Pull request by andrewbackes**
When using Redis to cache AMIs it is possible to receive a NullPointerException such as this:

```
java.lang.NullPointerException
 at com.netflix.spinnaker.orca.clouddriver.tasks.providers.aws.AmazonImageFinder$AmazonImage.toAmazonImageDetails(AmazonImageFinder.java:98)
 at com.netflix.spinnaker.orca.clouddriver.tasks.providers.aws.AmazonImageFinder.lambda$null$2(AmazonImageFinder.java:59)
 at java.util.Optional.map(Optional.java:215)
 at com.netflix.spinnaker.orca.clouddriver.tasks.providers.aws.AmazonImageFinder.lambda$byTags$3(AmazonImageFinder.java:59)
 at java.util.ArrayList.forEach(ArrayList.java:1249)
 at com.netflix.spinnaker.orca.clouddriver.tasks.providers.aws.AmazonImageFinder.byTags(AmazonImageFinder.java:56)
 at com.netflix.spinnaker.orca.clouddriver.tasks.image.FindImageFromTagsTask.execute(FindImageFromTagsTask.java:50)
 at com.netflix.spinnaker.orca.batch.TaskTasklet.doExecuteTask(TaskTasklet.groovy:196)
 ...
```

It all started by somebody deleting an AMI using the AWS Console. Somehow Spinnaker retained the cache entries keyed by AMI-name but removed entries keyed by AMI-id. Specifically, the `:relationships` entries had references to AMI id's that did not exist.

For example, this cache entry exists and refers to an ami-1234abcd:
```
com.netflix.spinnaker.clouddriver.aws.provider.AwsProvider:namedImages:relationships... = "[ \"aws:images:production:us-west-2:ami-1234abcd\" ]"
```
However, this cache entry does *not* exist:
```
com.netflix.spinnaker.clouddriver.aws.provider.AwsProvider:images:attributes:aws:images:production:us-west-2:ami-1234abcd
```

When this occurs the `/find` endpoint on clouddriver will have an empty `tagsByImageId` field. Ultimately that is what leads to the NullPointerException.

This PR will require an image to have a non-empty `tagsByImageId` field in order to be considered eligible by the 'Find Images by Tags' stage.

For reference the related code in clouddriver can be found [here AmazonNamedImageLookupController.groovy#L98](https://github.com/spinnaker/clouddriver/blob/4fc0c1e0f2a2ac89dc4e7cc9249407a12293edb6/clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/controllers/AmazonNamedImageLookupController.groovy#L98)

### CODE REVIEW

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/aws/AmazonImageFinder.java [hunk 1]
```diff
@@ -44,6 +44,7 @@
     List<AmazonImage> allMatchedImages =
         oortService.findImage(getCloudProvider(), packageName, null, null, prefixTags(tags)).stream()
         .map(image -> objectMapper.convertValue(image, AmazonImage.class))
+        .filter(image -> image.tagsByImageId.size() != 0)
```
Comment (lwander) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/aws/AmazonImageFinder.java]: Is it worth filtering on `null` `tagsByImageId` as well?

**Comment by lwander** [comment]
Any concerns if we merge @spinnaker/netflix-reviewers?

**Comment by anotherchrisberry** [comment]
LGTM - just needs a rebase and we'll get it merged

**Comment by andrewbackes** [comment]
Rebased

**Comment by anotherchrisberry** [comment]
Sorry, can you rebase one more time?

**Pull request merged**